src/jit/morph.cpp

```diff
@@ -5411,10 +5417,17 @@ void Compiler::fgMakeOutgoingStructArgCopy(
     // to ref counting of the lclVars.
     lvaTable[tmp].incRefCnts(compCurBB->getBBWeight(this), this);
 
-    GenTreePtr src;
     if (argx->gtOper == GT_OBJ)
     {
         argx->gtFlags &= ~(GTF_ALL_EFFECT) | (argx->AsBlk()->Addr()->gtFlags & GTF_ALL_EFFECT);
+        if (argx->OperMayThrow(this))
```

Comment:
I'm not usually a fan of `?:` but sometimes it seems useful: `argx->gtFlags |= argx->OperMayThrow(this) ? GTF_EXCEPT : GTF_IND_NONFAULTING;`